p2p/src/sync/tests/block_response.rs

```diff
@@ -15,294 +15,104 @@
-    T: NetworkingService + 'static,
-    T::ConnectivityHandle: ConnectivityService<T>,
-    T::SyncingMessagingHandle: SyncingMessagingService<T>,
-{
-    let addr = A::make_address();
-    let peer_id = P::new();
-
-    let config = Arc::new(common::chain::config::create_unit_test_config());
-    let (mut mgr, _conn, _sync, _pm) = make_sync_manager::<T>(A::make_transport(), addr).await;
-    register_peer(&mut mgr, peer_id).await;
-
-    let blocks = p2p_test_utils::create_n_blocks(
-        Arc::clone(&config),
-        TestBlockInfo::from_genesis(config.genesis_block()),
-        1,
+    let block = create_block(
+        Arc::clone(&chain_config),
+        TestBlockInfo::from_genesis(chain_config.genesis_block()),
     );
+    handle.send_message(peer, SyncMessage::BlockResponse(BlockResponse::new(block)));
 
-    let first = blocks[0].header().clone();
-    mgr.peers
-        .get_mut(&peer_id)
-        .unwrap()
-        .set_state(peer::PeerSyncState::UploadingBlocks(first.get_id()));
-
+    let (adjusted_peer, score) = handle.adjust_peer_score_event().await;
+    assert_eq!(peer, adjusted_peer);
     assert_eq!(
-        mgr.validate_block_response(&peer_id, blocks).await,
-        Ok(None),
+        score,
+        P2pError::ProtocolError(ProtocolError::UnexpectedMessage("")).ban_score()
     );
 }
 
 #[tokio::test]
-async fn valid_block_tcp() {
-    valid_block::<TestTransportTcp, PeerId, DefaultNetworkingService<TcpTransportSocket>>().await;
-}
-
-#[tokio::test]
-async fn valid_block_channels() {
-    valid_block::<TestTransportChannel, PeerId, DefaultNetworkingService<MpscChannelTransport>>()
-        .await;
-}
-
-#[tokio::test]
-async fn valid_block_noise() {
-    valid_block::<TestTransportNoise, PeerId, DefaultNetworkingService<NoiseTcpTransport>>().await;
-}
+async fn valid_response() {
+    let chain_config = Arc::new(create_unit_test_config());
+    let mut handle = SyncManagerHandle::with_config(Arc::clone(&chain_config)).await;
 
-// submit valid block
-async fn valid_block_invalid_state<A, P, T>()
-where
-    A: TestTransportMaker<Transport = T::Transport, Address = T::Address>,
-    P: MakeTestPeerId<PeerId = T::PeerId>,
-    T: NetworkingService + 'static,
-    T::ConnectivityHandle: ConnectivityService<T>,
-    T::SyncingMessagingHandle: SyncingMessagingService<T>,
-{
-    let addr = A::make_address();
-    let peer_id = P::new();
+    let peer = PeerId::new();
+    handle.connect_peer(peer).await;
 
-    let config = Arc::new(common::chain::config::create_unit_test_config());
-    let (mut mgr, _conn, _sync, _pm) = make_sync_manager::<T>(A::make_transport(), addr).await;
-    register_peer(&mut mgr, peer_id).await;
-
-    let blocks = p2p_test_utils::create_n_blocks(
-        Arc::clone(&config),
-        TestBlockInfo::from_genesis(config.genesis_block()),
-        1,
+    let blocks = create_n_blocks(
+        Arc::clone(&chain_config),
+        TestBlockInfo::from_genesis(chain_config.genesis_block()),
+        2,
```

Comment:
Why not a random number?

Reply:
I'm not sure what randomness adds here. The test checks that all requested blocks are sent and in my understanding any number more than 1 proves the point.

Reply:
You think it proves the point, until it finds a bug you didn't expect :wink:

The idea of randomizing tests is that we don't test an arbitrary case, but we test a model that makes us understand the code better and find bugs we never imagined.

In other words: You think that your point is proven with anything more than `1`? Great. Make a test that tests this assumption. There's no loss here. Only gain.

This principle should be applied everywhere in our code, because mistakes and oversights happen.